tool/tbot/botfs/botfs_test.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package botfs
+
+import (
+	"bytes"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestReadWrite attempts to test read/write against all possible symlink
+// modes.
+func TestReadWrite(t *testing.T) {
+	dir := t.TempDir()
+
+	secureWriteExpected, err := HasSecureWriteSupport()
+	require.NoError(t, err)
+
+	expectedData := []byte{1, 2, 3, 4}
+
+	tests := []struct {
+		mode SymlinksMode
+	}{
+		{SymlinksInsecure},
+		{SymlinksTrySecure},
+		{SymlinksSecure},
+	}
+
+	for _, test := range tests {
+		if test.mode == SymlinksSecure && !secureWriteExpected {
```

Comment:
```suggestion

	for _, mode := range []SymlinksMode{SymlinksInsecure, SymlinksTrySecure, SymlinksSecure}, {
		if mode == SymlinksSecure && !secureWriteExpected {
```

Seems simpler to just loop over the modes rather than an anonymous struct with 1 field.